library/common/jni/jni_utility.cc

```diff
@@ -279,39 +279,49 @@ jobjectArray ToJavaArrayOfByteArray(JNIEnv* env, const std::vector<std::string>&
 
 jbyteArray ToJavaByteArray(JNIEnv* env, const uint8_t* bytes, size_t len) {
   jbyteArray byte_array = env->NewByteArray(len);
-
-  env->SetByteArrayRegion(byte_array, 0, len, reinterpret_cast<const jbyte*>(bytes));
-
+  const jbyte* jbytes = reinterpret_cast<const jbyte*>(bytes);
+  env->SetByteArrayRegion(byte_array, /*start=*/0, len, jbytes);
   return byte_array;
 }
 
 void JavaArrayOfByteArrayToStringVector(JNIEnv* env, jobjectArray array,
                                         std::vector<std::string>* out) {
   size_t len = env->GetArrayLength(array);
   out->resize(len);
+
   for (size_t i = 0; i < len; ++i) {
     jbyteArray bytes_array = static_cast<jbyteArray>(env->GetObjectArrayElement(array, i));
     jsize bytes_len = env->GetArrayLength(bytes_array);
-    jbyte* bytes = env->GetByteArrayElements(bytes_array, nullptr);
+    // We don't care whether the returned array is a copy or not as we're simply
+    // copying it into C++ owned memory.
```

Comment:
I get what this comment is saying and it makes sense. But it threw me for a loop because it talks about copy twice, first to say we don't care and secondly to say we're copying. Of course, the first time is talking about GetByteArrayElements() and the second is talking about this method, avaArrayOfByteArrayToStringVector. So how about a slight tweak:

 // It doesn't matter if the array returned by GetByteArrayElements is a copy or not,
 // as the data will be simply be copied into C++ owned memory below.